Source/JavaScriptCore/API/JSValueRef.h

```diff
@@ -292,6 +320,140 @@ JS_EXPORT JSValueRef JSValueMakeString(JSContextRef ctx, JSStringRef string);
+    @param value      The JSValue to convert.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A uint64_t with the result of conversion, or 0 if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first. Then apply `ToBigUint64` if the numeric is a BigInt, otherwise do the truncation similar to `ToUint32` but for int64.
+*/
+JS_EXPORT uint64_t JSValueToUInt64(JSContextRef ctx, JSValueRef value, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Converts a JSValue to a singed 64-bit integer and returns the resulting integer.
+    @param ctx        The execution context to use.
+    @param value      The JSValue to convert.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A int64_t with the result of conversion, or 0 if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first. Then apply `ToBigInt64` if the numeric is a BigInt, otherwise do the truncation similar to `ToInt32` but for int64.
+*/
+JS_EXPORT int64_t JSValueToInt64(JSContextRef ctx, JSValueRef value, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Converts a JSValue to an unsigned 32-bit integer and returns the resulting integer.
+    @param ctx        The execution context to use.
+    @param value      The JSValue to convert.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A uint32_t with the result of conversion, or 0 if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first. Then apply `ToBigUint64` and truncate it to a uint32_t if the numeric is a BigInt, otherwise do `ToUint32`.
+*/
+JS_EXPORT uint32_t JSValueToUInt32(JSContextRef ctx, JSValueRef value, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Converts a JSValue to a singed 32-bit integer and returns the resulting integer.
+    @param ctx        The execution context to use.
+    @param value      The JSValue to convert.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A int32_t with the result of conversion, or 0 if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first. Then apply `ToBigInt64` and truncate it to a int32_t if the numeric is a BigInt, otherwise do `ToInt32`.
```

Comment:
The JSValue is converted to an integer according to the rules specified by the JavaScript language. If the JSValue is a BigInt then the value is truncated to an `int32_t`.